src/store/PeopleContext.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  ReactNode,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useFilters } from './FilterContext';
+
+type PeopleContextType = {
+  isLoading: boolean;
+  errorMsg: string;
+  filteredPeople: Person[];
+};
+
+export const PeopleContext = React.createContext<PeopleContextType | undefined>(
+  undefined,
+);
+
+type Props = {
+  children: ReactNode;
+};
+
+export const PeopleProvider: React.FC<Props> = ({ children }) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+  const { query, sexs, centuries, sort, order } = useFilters();
+
+  const filteredPeople = useMemo(() => {
+    let preparedPeople = [...people];
+
+    if (query) {
+      const lowerCaseQuery = query.toLowerCase();
+
+      preparedPeople = preparedPeople.filter(person =>
+        person.name.toLowerCase().includes(lowerCaseQuery),
+      );
+    }
+
+    if (sexs.length > 0) {
+      preparedPeople = preparedPeople.filter(person =>
+        sexs.includes(person.sex),
+      );
+    }
+
+    if (centuries.length > 0) {
+      preparedPeople = preparedPeople.filter(person => {
+        const personCentury = Math.ceil(person.born / 100);
+
+        return centuries.includes(personCentury.toString());
+      });
+    }
+
+    if (sort) {
+      preparedPeople.sort((a, b) => {
+        const valueA = a[sort as keyof Person];
+        const valueB = b[sort as keyof Person];
+
+        if (valueA === null || valueB === null) {
+          return 0;
```

Comment:
The sorting logic currently returns 0 if either `valueA` or `valueB` is `null`. This might not be the desired behavior if you want to handle `null` values differently, such as placing them at the end of the list. Consider adjusting this logic if needed.